core/src/main/java/org/elasticsearch/index/mapper/AllFieldMapper.java

```diff
@@ -112,7 +111,7 @@ public AllFieldMapper build(BuilderContext context) {
             // the AllFieldMapper ctor in the builder since it is not valid. Here we validate
             // the doc values settings (old and new) are rejected
             Object docValues = node.get("doc_values");
-            if (docValues != null && lenientNodeBooleanValue(docValues)) {
+            if (docValues != null && TypeParsers.nodeBooleanValue(name, "doc_values", docValues, parserContext)) {
```

Comment:
Do we need lenient BWC here? I think we might. I know mappings are converted on upgrade but I wonder if we expect people to be able to send `off` to old indexes. I think it is reasonable to tell them that they can't but it is weird to explain the you can for configuring analyzers and can't for mappings....